frontend/packages/console-app/src/components/modals/clone/clone-pvc-modal.tsx

```diff
@@ -3,55 +3,69 @@ import './_clone-pvc-modal.scss';
 import * as React from 'react';
 
 import { Form, FormGroup, TextInput } from '@patternfly/react-core';
-import { K8sResourceKind, k8sCreate } from '@console/internal/module/k8s';
+import { K8sResourceKind, k8sCreate, referenceFor } from '@console/internal/module/k8s';
 import {
   LoadingInline,
   ResourceIcon,
   humanizeBinaryBytes,
+  history,
   withHandlePromise,
-} from '@console/internal/components/utils/index';
+  RequestSizeInput,
+  validate,
+  resourceObjPath,
+} from '@console/internal/components/utils';
 import {
   ModalBody,
   ModalComponentProps,
   ModalSubmitFooter,
   ModalTitle,
   createModalLauncher,
 } from '@console/internal/components/factory';
-
-import { DataPoint } from '@console/internal/components/graphs/index';
+import { DataPoint } from '@console/internal/components/graphs';
 import { HandlePromiseProps } from '@console/internal/components/utils/promise-component';
-import { PersistentVolumeClaimModel } from '@console/internal/models/index';
+import { PersistentVolumeClaimModel } from '@console/internal/models';
 import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
 import { getInstantVectorStats } from '@console/internal/components/graphs/utils';
-import { getPVCUsedCapacityQuery } from '../../../constants/queries';
 import { usePrometheusPoll } from '@console/internal/components/graphs/prometheus-poll-hook';
+import { getRequestedPVCSize } from '@console/shared/src/selectors';
 
 const accessModeLabels = Object.freeze({
   ReadWriteOnce: 'Single User (RWO)',
   ReadWriteMany: 'Shared Access (RWX)',
   ReadOnlyMany: 'Read Only (ROX)',
 });
 
+const dropdownUnits = {
+  Mi: 'MiB',
+  Gi: 'GiB',
+  Ti: 'TiB',
+};
+
 const ClonePVCModal = withHandlePromise((props: ClonePVCModalProps) => {
   const { close, cancel, resource, handlePromise, errorMessage, inProgress } = props;
-  const pvcName: string = resource.metadata.name;
-  const [clonePVCName, setClonePVCName] = React.useState(`${pvcName}-clone`);
+  const { name: pvcName, namespace } = resource?.metadata;
+  const defaultSize: string[] = validate.split(getRequestedPVCSize(resource));
 
-  const pvcUsedCapacityQuery: string = getPVCUsedCapacityQuery(pvcName);
+  const [clonePVCName, setClonePVCName] = React.useState<string>(`${pvcName}-clone`);
+  const [requestedSize, setRequestedSize] = React.useState<string>(defaultSize[0] || '');
+  const [requestedUnit, setRequestedUnit] = React.useState<string>(defaultSize[1] || 'Gi');
```

Comment:
nit: the string type is implicit from the initial value, no need to pass the type.